janusgraph-core/src/main/java/org/janusgraph/diskstorage/locking/consistentkey/ConsistentKeyLocker.java

```diff
@@ -382,13 +382,30 @@ private WriteResult tryWriteLockOnce(StaticBuffer key, StaticBuffer del, StoreTr
         final Timer writeTimer = times.getTimer().start();
         StaticBuffer newLockCol = serializer.toLockCol(writeTimer.getStartTime(), rid, times);
         Entry newLockEntry = StaticArrayEntry.of(newLockCol, zeroBuf);
+        StoreTransaction newTx = null;
         try {
-            final StoreTransaction newTx = overrideTimestamp(txh, writeTimer.getStartTime());
+            newTx = overrideTimestamp(txh, writeTimer.getStartTime());
+
             store.mutate(key, Collections.singletonList(newLockEntry),
                 null == del ? KeyColumnValueStore.NO_DELETIONS : Collections.singletonList(del), newTx);
+
+            newTx.commit();
+            newTx = null;
         } catch (BackendException e) {
             log.debug("Lock write attempt failed with exception", e);
             t = e;
+        } finally {
+            if (newTx != null) {
+                try {
+                    if (log.isDebugEnabled()) {
+                        log.debug("Transaction is still open! Rolling back", new Throwable());
```

Comment:
Why do we adding `new Throwable()` to the log? I guess, it isn't useful because it doesn't have any information. Instead, I would log `toString()` of the transaction, which may have more useful information. I.e. something like:
```
if (log.isDebugEnabled()) {
    log.debug("Transaction is still open. Rolling back the transaction: "+tx.toString());
}
```

Reply:
Throwable in this debug log statement will print the stack, which could be useful in troubleshooting leaked transactions. Let me know if this should be removed.

I agree on including tx.toString() in the log. Will update shortly for this.

Reply:
Good point. Agree about stack trace. It might be useful